scarlet/component.py

```diff
@@ -525,19 +539,36 @@ def get_model(self, *params):
         model: array
             (Bands, Height, Width) data cube
         """
-        model = np.zeros(self.frame.shape)
-        if len(params):
-            i = 0
-            for k, c in enumerate(self.components):
+        full_model = np.zeros(self.frame.shape, dtype=self.frame.dtype)
+
+        models = []
+        slices = []
+        i = 0
+
+        for k, c in enumerate(self.components):
+            if len(params):
                 j = len(c.parameters)
-                p = params[i : i + j]
+                p = params[i: i + j]
                 i += j
-                model = model + c.get_model(*p)
-        else:
-            for c in self.components:
-                model = model + c.get_model()
+                model = c.get_model(*p)
+            else:
+                model = c.get_model()
+
+            models.append(model)
+
+            # Get the slices needed to insert the model
+            imbox = Box.from_image(full_model)
+            subbox = Box.from_image(model)
+            imbox -= c.bbox.origin
+            overlap = imbox & subbox
+            slices.append((c.bbox.slices_for(full_model), overlap.slices_for(model)))
+
+        full_model = _add_models(*models, full_model=full_model, slices=slices)
```

Comment:
You are calling the function directly here, I think it's correct to also define the gradient primitive here instead of in `Blend`